libs/ledger-live-common/src/families/filecoin/bridge/utils/erc20/tokenAccounts.ts

```diff
@@ -0,0 +1,179 @@
+
+    invariant(ops, "filecoin operation is not defined");
+
+    return ops;
+  } catch (e) {
+    log("error", "filecoin error converting erc20 transaction to operation", e);
+    return [];
+  }
+};
+
+export async function buildTokenAccounts(
+  filAddr: string,
+  parentAccountId: string,
+  initialAccount?: Account,
+): Promise<TokenAccount[]> {
+  try {
+    const nativeEthAddr = await convertAddressFilToEth(filAddr);
+    const transfers = await fetchERC20Transactions(nativeEthAddr);
+    const transfersUntangled: { [addr: string]: ERC20Transfer[] } = transfers.reduce(
+      (prev, curr) => {
+        curr.contract_address = curr.contract_address.toLowerCase();
+        if (prev[curr.contract_address]) {
+          prev[curr.contract_address] = [...prev[curr.contract_address], curr];
+        } else {
+          prev[curr.contract_address] = [curr];
+        }
+        return prev;
+      },
+      {},
+    );
+
+    const subs: TokenAccount[] = [];
+    for (const [cAddr, txns] of Object.entries(transfersUntangled)) {
+      const token = findTokenByAddressInCurrency(cAddr, "filecoin");
+      if (!token) {
+        log("error", `filecoin token not found, addr: ${cAddr}`);
+        continue;
+      }
+
+      const balance = await fetchERC20TokenBalance(nativeEthAddr, cAddr);
+      const bnBalance = new BigNumber(balance.toString());
+      const tokenAccountId = encodeTokenAccountId(parentAccountId, token);
+
+      const operations = txns
+        .flatMap(txn => erc20TxnToOperation(txn, nativeEthAddr, tokenAccountId, token.units[0]))
+        .flat();
+
+      if (operations.length === 0 && bnBalance.isZero()) {
+        continue;
+      }
+
+      const maybeExistingSubAccount =
+        initialAccount &&
+        initialAccount.subAccounts &&
+        initialAccount.subAccounts.find(a => a.id === tokenAccountId);
+
+      const sub: TokenAccount = {
+        type: "TokenAccount",
+        id: tokenAccountId,
+        parentId: parentAccountId,
+        token,
+        balance: bnBalance,
+        spendableBalance: bnBalance,
+        operationsCount: txns.length,
+        operations,
+        pendingOperations: maybeExistingSubAccount ? maybeExistingSubAccount.pendingOperations : [],
+        creationDate: operations.length > 0 ? operations[0].date : new Date(),
+        swapHistory: maybeExistingSubAccount ? maybeExistingSubAccount.swapHistory : [],
+        balanceHistoryCache: emptyHistoryCache, // calculated in the jsHelpers
+      };
+
+      subs.push(sub);
+    }
+
+    return subs;
+  } catch (e) {
+    log("error", "filecoin error building token accounts", e);
+    return [];
+  }
+}
+
+export const encodeTxnParams = (abiEncodedParams: string) => {
+  if (!abiEncodedParams) {
+    throw new Error("Cannot encode empty abi encoded params");
+  }
+
+  const buffer = Buffer.from(abiEncodedParams.slice(2), "hex"); // buffer/byte array
+  const dataEncoded = cbor.encode(buffer);
+
+  return base64.stringify(dataEncoded);
```

Comment:
I think we can use
Buffer.from
and
buffer.toString('base64');

instead, so that we don't have to use "rfc4648" lib

Reply:
Great tip, I didn't know it could be done this way. Now using this way and the `rfc4648` is removed from the package.json. 😄